plugin-locationcomponent/src/main/java/com/mapbox/maps/plugin/locationcomponent/LocationCompassEngine.kt

```diff
@@ -0,0 +1,300 @@
+package com.mapbox.maps.plugin.locationcomponent
+
+import android.content.Context
+import android.hardware.Sensor
+import android.hardware.SensorEvent
+import android.hardware.SensorEventListener
+import android.hardware.SensorManager
+import android.os.SystemClock
+import android.view.Surface
+import android.view.WindowManager
+import com.mapbox.common.Logger
+
+/**
+ * This class handles compass events and tracking the current device heading.
+ */
+internal class LocationCompassEngine(context: Context) : SensorEventListener {
```

Comment:
nit : just `CompassEngine` ?

Reply:
Yes, this is the name for V9, do you have another suggested name?

Reply:
No, we can use any name, but this class is referred from the one in [v9](https://github.com/mapbox/mapbox-gl-native-android/blob/main/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/location/LocationComponentCompassEngine.java)